fiftyone/utils/csv.py

```diff
@@ -0,0 +1,215 @@
+                subfolder of ``export_dir`` in which to export the media
+            -   an absolute directory path in which to export the media. In
+                this case, the ``export_dir`` has no effect on the location of
+                the data
+            -   a JSON filename like ``"data.json"`` specifying the filename of
+                the manifest file in ``export_dir`` generated when
+                ``export_media`` is ``"manifest"``
+            -   an absolute filepath specifying the location to write the JSON
+                manifest file when ``export_media`` is ``"manifest"``. In this
+                case, ``export_dir`` has no effect on the location of the data
+
+            If None, the default value of this parameter will be chosen based
+            on the value of the ``export_media`` parameter
+        labels_path (None): an optional parameter that enables explicit control
+            over the location of the exported labels. Can be any of the
+            following:
+
+            -   a filename like ``"labels.csv"`` specifying the location in
+                ``export_dir`` in which to export the labels
+            -   an absolute filepath to which to export the labels. In this
+                case, the ``export_dir`` has no effect on the location of the
+                labels
+
+            If None, the labels will be exported into ``export_dir`` using the
+            default filename
+        export_media (None): controls how to export the raw media. The
+            supported values are:
+
+            -   ``True``: copy all media files into the output directory
+            -   ``False``: don't export media
+            -   ``"move"``: move all media files into the output directory
+            -   ``"symlink"``: create symlinks to the media files in the output
+                directory
+            -   ``"manifest"``: create a ``data.json`` in the output directory
+                that maps UUIDs used in the labels files to the filepaths of
+                the source media, rather than exporting the actual media
+
+            If None, the default value of this parameter will be chosen based
+            on the value of the ``data_path`` parameter
+        rel_dir (None): an optional relative directory to strip from each input
+            filepath to generate a unique identifier for each media. When
+            exporting media, this identifier is joined with ``data_path`` to
+            generate an output path for each exported media. This argument
+            allows for populating nested subdirectories that match the shape of
+            the input paths. The path is converted to an absolute path (if
+            necessary) via :func:`fiftyone.core.utils.normalize_path`
+        abs_paths (False): whether to store absolute paths to the media in the
+            exported labels
+        media_path ("filepath"): the name of the field containing the media to
```

Comment:
Propose rename (refactor to whole file)
```suggestion
        media_path_field ("filepath"): the name of the field containing the media to
```

Reply:
I actually meant to use `media_field`. What do you think about that instead? If you grep the codebase (especially fiftyone-teams), you'll see a variety of uses of `media_field="filepath"` to allow the user to customize what field contains the media to export.

Reply:
> Approved but with questions/suggestions

Thanks for the review! FYI I need to add a couple more boilerplate things to this PR before it is mergable:
- Document the new export type here: https://github.com/voxel51/fiftyone/blob/develop/docs/source/user_guide/export_datasets.rst
- Add unit tests here: https://github.com/voxel51/fiftyone/blob/develop/tests/unittests/import_export_tests.py

Reply:
Yep `media_field` sounds good to me!

I can add in those other pieces? unless you have done them already 😉